cabal-install/src/Distribution/Client/ProjectConfig/Legacy.hs

```diff
@@ -1076,6 +1079,17 @@ legacyPackageConfigFieldDescrs =
           dispFlagAssignment parsecFlagAssignment
           configConfigurationsFlags
           (\v conf -> conf { configConfigurationsFlags = v })
+      , simpleFieldParsec "dump-buildinfo"
+          (fromFlagOrDefault Disp.empty . fmap (\bi -> Disp.text $ case bi of
+            DumpBuildInfo -> "True"
+            NoDumpBuildInfo -> "False"))
+          (toFlag <$> do
+            b <- parsec
+            if b then pure DumpBuildInfo else pure NoDumpBuildInfo
+            <|> pure mempty
+            )
+          configDumpBuildInfo
+          (\v conf -> conf { configDumpBuildInfo = v })
```

Comment:
This is intentionally terrible so someone can tell me how to improve it and where it should be located